### PR TITLE
Follow a user via PUT

### PR DESCRIPTION
**Pull request by manuelq12**
New test to update a users information by following a user

### CODE REVIEW

File: test/features/support/common/body.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const { expect } = require('chai');
+const { Then } = require('cucumber');
+
+Then('the response body must be empty', function () {
+  expect(this.response.body).to.eql({});
```
Comment (yesid-lopez) [inline test/features/support/common/body.js]: chai has a better assertion than `.to.eql({})`. Please try to improve it

File: test/features/support/common/statusCode.js [hunk 2]
```diff
@@ -5,3 +5,7 @@ const { Then } = require('cucumber');
 Then('the response should contain a OK status', function () {
   expect(this.response.status).to.equal(statusCode.OK);
 });
+
+Then('the response should contain a NO CONTENT status', function () {
```
Comment (yesid-lopez) [inline test/features/support/common/statusCode.js]: you can create a more dynamic step to reuse in general situations

File: test/features/support/common/body.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+const { expect } = require('chai');
+const { Then } = require('cucumber');
+
+Then('the response body must be empty', function () {
+  /* eslint-disable no-unused-expressions */
```
Comment (yesid-lopez) [inline test/features/support/common/body.js:5]: can you add a rule for that?

File: test/features/support/common/statusCode.js [hunk 4]
```diff
@@ -1,7 +1,15 @@
-const statusCode = require('http-status-codes');
 const { expect } = require('chai');
 const { Then } = require('cucumber');
 
-Then('the response should contain a OK status', function () {
-  expect(this.response.status).to.equal(statusCode.OK);
+Then('the response should contain a {int} status code', function (statusCodeExpected) {
+  switch (statusCodeExpected) {
```
Comment (yesid-lopez) [inline test/features/support/common/statusCode.js]: why do you need a switch case ?

File: package.json [hunk 5]
```diff
@@ -33,7 +33,6 @@
     "eslint-config-airbnb-base": "^14.2.1",
     "eslint-plugin-cucumber": "^1.4.0",
     "eslint-plugin-import": "^2.22.1",
-    "http-status-codes": "^2.1.4",
```
Comment (aperdomob) [inline package.json]: why did you delete this library?

File: .eslintrc.yml [hunk 6]
```diff
@@ -11,3 +11,4 @@ rules:
   "comma-dangle": ["error", "never"]
   func-names: ["error", "never"]
   linebreak-style: ["error", "unix"]
+  no-unused-expressions: 0
```
Comment (aperdomob) [inline .eslintrc.yml]: I disagree with this change, put in 0 this property removes several rules. https://eslint.org/docs/rules/no-unused-expressions

Also, you only disable that to Chai asserts then disable directly into the test (like before Yesid comment). or search a library in npm with Chai eslint rules
@leonleo997

File: test/features/GithubApi.Authentication.feature [hunk 7]
```diff
@@ -3,7 +3,7 @@ Feature: Github Api Authentication Test
         Given a Github username "<username>"
         And the repository of the Github user "<repository>"
         When the user tries to retrieve its repository with OAuth2 Tokens by Header
-        Then the response should contain a OK status
+        Then the response should contain a 200 status code
```
Comment (aperdomob) [inline test/features/GithubApi.Authentication.feature]: use the text instead of the number, the text has more readability

File: test/agent/Agent.js [hunk 8]
```diff
@@ -0,0 +1,67 @@
+const agent = require('superagent');
+
+class Agent {
```
Comment (aperdomob) [inline test/agent/Agent.js]: Agent is a weird name for that class and it's confusing with SuperAgent, avoid using the same names that your imported libraries because you are creating dependencies. What happens if you use another library and it doesn't have the Agent concept?

File: test/agent/Agent.js [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+  static async getRequest(url) {
+    const response = await agent.get(url)
+      .auth('token', process.env.ACCESS_TOKEN)
+      .set('User-Agent', 'agent');
+
+    return response;
```
Comment (aperdomob) [inline test/agent/Agent.js]: every method is very similar and your goal is encapsule the authorization and headers, please check this library https://www.npmjs.com/package/superagent-defaults

**Review by aperdomob — CHANGES_REQUESTED** [review]

File: package.json (hunk 5 above)
Comment (manuelq12) [reply package.json]: Since we were using the status code and not it's meaning, we could give the code expected directly from the feature, so using the  http-status-code was redundant

**Review by yesid-lopez — APPROVED** [review]
